handler/src/test/java/io/netty/handler/ssl/SSLEngineTest.java

```diff
@@ -357,6 +383,111 @@ public void testSSLSessionId() throws Exception {
+                                    ctx.writeAndFlush(ctx.alloc().buffer(1).writeByte(100));
+                                }
+                                ctx.fireUserEventTriggered(evt);
+                            }
+
+                            @Override
+                            public void channelRead(final ChannelHandlerContext ctx, Object msg) {
+                                ReferenceCountUtil.release(msg);
+                                // The server then attempts to trigger a flush operation once the application data is
+                                // received from the client. The flush will encrypt all data and should not result in
+                                // deadlock.
+                                ctx.channel().eventLoop().schedule(new Runnable() {
+                                    @Override
+                                    public void run() {
+                                        ctx.writeAndFlush(ctx.alloc().buffer(1).writeByte(101));
+                                    }
+                                }, 500, TimeUnit.MILLISECONDS);
+                            }
+
+                            @Override
+                            public void channelInactive(ChannelHandlerContext ctx) {
+                                serverLatch.countDown();
+                            }
+                        });
+                        serverConnectedChannel = ch;
+                    }
+                });
+
+        serverChannel = sb.bind(new InetSocketAddress(0)).syncUninterruptibly().channel();
+
+        clientSslCtx = SslContextBuilder.forClient()
+                .sslProvider(SslProvider.JDK) // OpenSslEngine doesn't support renegotiation on client side
+                .trustManager(InsecureTrustManagerFactory.INSTANCE).build();
+
+        cb = new Bootstrap();
+        cb.group(new NioEventLoopGroup(1))
+                .channel(NioSocketChannel.class)
+                .handler(new ChannelInitializer<SocketChannel>() {
+                    @Override
+                    public void initChannel(SocketChannel ch) {
+                        ChannelPipeline p = ch.pipeline();
+                        SslHandler sslHandler = clientSslCtx.newHandler(ch.alloc());
+                        // The renegotiate is not expected to succeed, so we should stop trying in a timely manner so
+                        // the unit test can terminate relativley quicly.
+                        sslHandler.setHandshakeTimeout(1, TimeUnit.SECONDS);
+                        p.addLast(sslHandler);
+                        p.addLast(new ChannelInboundHandlerAdapter() {
+                            private int handshakeCount;
+                            @Override
+                            public void userEventTriggered(ChannelHandlerContext ctx, Object evt) {
+                                // OpenSSL SSLEngine sends a fatal alert for the renegotiation handshake because the
```

Comment:
@normanmaurer - test failure that was observed for JDK is explained here ... ptal